OrbitCore/TimerManager.cpp

```diff
@@ -47,8 +47,8 @@ void TimerManager::StartRecording() {
     return;
   }
 
-  if (!m_ConsumerThread) {
-    m_ConsumerThread = new std::thread([&]() { ConsumeTimers(); });
+  if (!consumerThread_.joinable()) {
```

Comment:
Seems like this should be a CHECK... we already have a guard, if this called in in MT environment this check itself produces the race, so it does not prevent from creating multiple threads if called concurrently.

Reply:
yes. sounds reasonable. done.